pyccel/ast/cwrapper.py

```diff
@@ -525,12 +555,53 @@ def pybool_to_bool(cast_function_name):
                        body      = cast_function_body,
                        results   = [cast_function_result])
 
+def pyarray_to_ndarray(cast_function_name):
+    cast_function_argument = Variable(dtype=PyccelPyArrayObject(), name='o', is_pointer=True)
+    cast_function_result   = Variable(dtype=PyccelPyArrayObject(), name = 'c', rank=1)
+
+    nd          = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='nd'))
+    raw_data    = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='raw_data'))
+    shape       = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='shape'))
+    type_size   = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='type_size'))
+    strides     = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='strides', allocatable=True))
+    arr_type    = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeGeneric(), name='type'))
+    length      = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='length'))
+    buffer_size = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeInteger(), name='buffer_size'))
+    is_view    = DottedVariable(cast_function_result,
+                        Variable(dtype=NativeBool(), name='is_view'))
+
+    cast_function_body = [Assign(nd, FunctionCall(numpy_get_ndims, [cast_function_argument])),
+                          Assign(raw_data, FunctionCall(numpy_get_data, [cast_function_argument])),
+                          Assign(shape, FunctionCall(numpy_get_shape, [cast_function_argument])),
```

Comment:
Do the types match here?

Reply:
now yes.

Reply:
On my computer numpy shapes are stored with type `npy_intp`. According to [this](https://numpy.org/doc/1.17/reference/c-api.dtype.html#c.NPY_INTP), it is platform dependant. As we always used fixed precision we are not platform dependant. I think it would be safer to copy, unless I'm looking at the wrong line?

Reply:
I agree with you, I see that `npy_intp`  is a platform dependant.

Reply:
do the tests passed in your machine?

Reply:
They pass on my machine, and they seem to pass on windows too according to github which is a good sign. But I still think a copy would be better. You never know if it could change in the future and that way all platforms should work.

Plus if we copy the shape then we don't need the strange logic to call `free` we can just use the garbage collector implemented and treat the object as a pointer

Reply:
I wonder how you did solve this problem in Fortran?